### PR TITLE
Cleanup CAS attributes

### PR DESCRIPTION
**Pull request by MarcoRossignoli**
Removed dead code.

/cc @jkotas

### CODE REVIEW

**Comment by MarcoRossignoli** [comment]
@jkotas  found also 

```
λ findstr /s /n /c:"[SecuritySafeCritical]" *.cs
Common\src\CoreLib\System\Runtime\CompilerServices\AsyncValueTaskMethodBuilder.cs:103:        [SecuritySafeCritical]
Common\src\CoreLib\System\Runtime\CompilerServices\AsyncValueTaskMethodBuilder.cs:210:        [SecuritySafeCritical]
Common\src\System\Net\Logging\NetEventSource.Common.cs:49:    [SecuritySafeCritical]
System.Diagnostics.DiagnosticSource\src\System\Diagnostics\Activity.cs:442:        [SecuritySafeCritical]
```

do i remove also these(with conditional code)?

File: src/System.Reflection.Context/src/System/Reflection/Context/Delegation/DelegatingAssembly.cs [hunk 1]
```diff
@@ -98,8 +98,7 @@ public override Type[] GetExportedTypes()
         {
             return UnderlyingAssembly.GetExportedTypes();
         }
-
-        [SecurityCritical]
+
```
Comment (bartonjs) [inline src/System.Reflection.Context/src/System/Reflection/Context/Delegation/DelegatingAssembly.cs]: It seems that the removal left all the spaces at the beginning of the line, so now it's a line of non-zero whitespace.

Comment (MarcoRossignoli) [reply src/System.Reflection.Context/src/System/Reflection/Context/Delegation/DelegatingAssembly.cs]: you're right

Comment (MarcoRossignoli) [reply src/System.Reflection.Context/src/System/Reflection/Context/Delegation/DelegatingAssembly.cs]: @bartonjs is there a way to see that on "web client"?

Comment (bartonjs) [reply src/System.Reflection.Context/src/System/Reflection/Context/Delegation/DelegatingAssembly.cs]: Highlight the diff region, the whitespace characters get an inverted background, but the "there's just screen here" doesn't.

Comment (MarcoRossignoli) [reply src/System.Reflection.Context/src/System/Reflection/Context/Delegation/DelegatingAssembly.cs]: it would be useful "View White Space" VS style.

**Comment by danmosemsft** [comment]
For the others, at least two and maybe all are in code that still ships for .NET Framework, so best left alone for now.

**Comment by danmosemsft** [comment]
@weshaggard for this code in the PR my assumption is that it (System.Reflection.Context) does not ship on NETFX since it had ` <InboxOnTargetFramework Include="net45">`

**Comment by weshaggard** [comment]
That is a correct assumption it is inbox on .NET Framework so we don't ship a version in the package.

**Comment by MarcoRossignoli** [comment]
@danmosemsft https://github.com/dotnet/corefx/issues/30188#issuecomment-408025286

**Pull request merged**